pkg/kubectl/cmd/get/get.go

```diff
@@ -552,7 +554,7 @@ func (o *GetOptions) Run(f cmdutil.Factory, cmd *cobra.Command, args []string) e
 			if lastMapping != nil && !o.NoHeaders {
 				// If we've written output since the last time we started a new set of headers, write an empty line to separate object types
 				if written != trackingWriter.Written {
```

Comment:
I think we can replace all of this with resetting the separatorWriter

```diff
- // If we've written output since the last time we started a new set of headers, write an empty line to separate object types
- if written != trackingWriter.Written {
-   fmt.Fprintln(w)
-   written = trackingWriter.Written
- }
+ separatorWriter.SetReady(true)
```

Reply:
That's neat. Thanks.

Updated the logic, comments and added a testcase. PTAL.